Azure getting started refinement (#167)
NordicSemiconductor/asset-tracker-cloud-docs: docs/azure/GettingStarted/Deploy.rst

      az provider register --namespace Microsoft.AzureActiveDirectory

#. Follow the `tutorial for creating an Azure Active Directory B2C <https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-create-tenant>`_. On the creation screen, select the resource group you have created above. Currently, it is not possible to create an Active Directory B2C and application through the ARM template (see `GitHub issue <https://github.com/NordicSemiconductor/asset-tracker-cloud-azure-js/issues/1>`_).

#. Save the initial domain name of the created Active Directory B2C to the environment variable ``B2C_TENANT``.
   In this example, ``nrfassettrackerusers`` is used as the initial domain name.

   .. code-block:: bash

      # add to .envrc
      export B2C_TENANT="nrfassettrackerusers"

> #. After creating the Active Directory B2C, under :guilabel:`User flows`, create the user flow for sign up and sign in (recommended version), and make sure to name the userflow as ``B2C_1_signup_signin``.

Reviewer: ```suggestion
#. Select :guilabel:`User flows`, create the user flow for sign up and sign in (recommended version), and make sure to name the userflow as ``B2C_1_signup_signin``.
```
Should we also mention where to find the User flows? Is it under Policies?

Author: It's on the left-hand side visible at the top level.
![image](https://user-images.githubusercontent.com/188915/119957570-7694c300-bfa2-11eb-9846-32b4c03f18de.png)

#. Follow the instructions in the `tutorial for registering a web application in Azure Active Directory B2C <https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-register-applications?tabs=app-reg-ga#register-a-web-application>`_ and register a web application. Use ``https://<your APP_NAME>app.z16.web.core.windows.net/`` as the redirect URL.

> #. After registering the application, under :guilabel:`Authentication`, enable the implicit grant and hybrid flows for :guilabel:`Access tokens` and :guilabel:`ID tokens` and click :guilabel:`Save`.

Reviewer: ```suggestion
#. Select :guilabel:`Authentication`, enable the implicit grant and hybrid flows for :guilabel:`Access tokens` and :guilabel:`ID tokens` and click :guilabel:`Save`.
```
Where is the Authentication exactly?

Author: ![image](https://user-images.githubusercontent.com/188915/119957681-95935500-bfa2-11eb-9400-8a136cd5d674.png)

#. Save the ``application (client) id`` to the environment variable ``APP_REG_CLIENT_ID`` in the :file:`.envrc` file:

   .. code-block:: bash

      # add to .envrc
      export APP_REG_CLIENT_ID=...

#. Grant the app registration directory API permissions for the function app:

   a. Click :guilabel:`Expose an API`.
   

>    #. Set the ``Application ID URI`` field to ``api``. Click :guilabel:`Save and continue`. 

Reviewer: ```suggestion
   #. Set the ``Application ID URI`` field to ``api`` and click :guilabel:`Save and continue`. 
```

      # Deploy the functions
      func azure functionapp publish ${APP_NAME:-nrfassettracker}API --typescript

   If the command gives an error, you can find the detailed log message using the printed tracking ID and the following command:

   .. parsed-literal::
      :class: highlight

      az monitor activity-log list --correlation-id "*tracking ID*" \\
         | jq '.[].properties.statusMessage | fromjson'

   It can take a few minutes for the detailed log message to be populated.

>    If the error message does not include a tracking ID navigate to the resource group in the Azure portal and review the deployments.
>    There should be a failed deployment called ``initial-setup``.

Reviewer: ```suggestion
   If the error message does not include a tracking ID, navigate to the resource group in the Azure portal and review the deployments.
   There is a failed deployment called ``initial-setup``.
```
